### ISSUE TITLE
Suggestion: parameters order in EntityAdapter methods

### ISSUE THREAD

**Issue by Arsync** [ForeverFast/ForeverFast.Fluxor.Extensions#2]
As we can see in Fluxor reducer methods, the first parameter is 'state' and the second parameter is 'action'.
```cs
[ReducerMethod]
public static ProductState OnProductsLoaded(
    ProductState state,
    ProductsLoadedAction action
)
    => Adapter.SetMany(state, action.Entities);
```
So why not to sort parameters of all EntityAdapter methods with the same order?
```cs
public TState SetMany<TState>(TState state, ICollection<TEntity> entities)
    where TState : EntityState<TEntity, TKey>
{
    var items = entities.Select(x => KeyValuePair.Create(_selectIdCallback(x), x));

    return state with
    {
        Entities = state.Entities.SetItems(items)
    };
}

public TState Map<TState>(TState state, TKey key, Func<TEntity, TEntity> mapFunc)
    where TState : EntityState<TEntity, TKey>
{
    return state with
    {
        Entities = state.Entities.SetItem(key, mapFunc(state.Entities[key]))
    };
}
```
Check this:
```cs
// Before
Adapter.Map(key, entity => entity with
{
    Name = action.Name
}, state);

// After
Adapter.Map(state, key, entity => entity with
{
    Name = action.Name
});
```

**Comment by Metalvast** [comment]
I agree, it looks more convenient. I'll fix it on the weekend. 

By the way, do you have any experience in creating an application state? Right now I'm looking for this for a work project where we use Fluxor, but using it causes some performance issues.

**Comment by Arsync** [comment]
> By the way, do you have any experience in creating an application state? Right now I'm looking for this for a work project where we use Fluxor, but using it causes some performance issues.

Depends on what is need to do. My application state includes user profile, for example.

**Comment by Metalvast** [comment]
Fixed in v0.0.4

> Depends on what is need to do.

I am interested in any general state building practices
